### DISCUSSION TITLE
[Announcement] Discussion about adding very limited statistics reporting on the instance info page

### DISCUSSION THREAD

**Discussion by sigaloid** [spikecodes/libreddit#807] [Polls]
Given the ongoing dilemma in #785 (Reddit API pricing changes), I'd like to rekindle the discussion from #624 and discuss adding some statistics collection to Libreddit. It's more relevant now because **it would make calculating the theoretical estimated cost of Reddit API usage much easier for instance operators**.

If added, it will be added to the **instance info page** (ie, we as maintainers don't implicitly get the stats to some server, they're public in your instance under `/instance-info`).

Stats I'm considering collecting:

* Count number of requests being made to Reddit (after caching)
* Count number of requests to the server total

Whether you are an instance operator or a user, please vote!

**Comment by Tokarak** [comment]
It's an interesting question on who should be the privacy gatekeeper — the maintainers or the instance hosters? I can see that maximum privacy is part of the identity of this project, and some level of trust would disappear if you can no longer trust the source code, but a few instances subject to change. On the other hand, liberalisation gives more freedom to both the maintainers and hosters, without removing the option of current maximal privacy. 
Logging is the right of a web-app to not live in a state of amnesia. We don't have to get into wether this trumps certain privacy rights of users, because in this case the logging consists of private matters between the instance and the reddit api. 
The only information this gives about users is a vague idea of the number of them (which can already be estimated by factors like eg memory usage), as well as some idea of the timezone of the user demographic. This may be a larger issue on smaller instances (note that this affects a small number of users); can be remedied by smaller instances disabling this functionality, if they want.
My more general suggestion is that optional anonymous statistics can fit in with this project, as they are not adverse to privacy. This means the approach to privacy needs to be carefully evaluated.
NB — I wonder if the "right to log" (a phrase I made up) is respected by privacy laws eg GDPR? Maybe a good privacy policy would involve dedication to not requiring to display a privacy or cookie notice on the site.

**Comment by Tokarak** [comment]
Alternative approach: the instance aggregates the number of requests when it loads a web page, sends it to the user, and immediately forgets about it. The user updates a counter in local storage. This will give the number of reddit requests per user, while being truly "opt-in" to share. 
This approach IMO is needlessly non-trivial.
